api/v1alpha1/cruisecontroloperation_types.go

```diff
@@ -93,3 +95,112 @@ type CruiseControlTask struct {
 func init() {
 	SchemeBuilder.Register(&CruiseControlOperation{}, &CruiseControlOperationList{})
 }
+
+func (task *CruiseControlTask) SetDefaults() {
+	task.Finished = nil
+	task.State = ""
+	task.Started = nil
+	task.ErrorMessage = ""
+	task.HTTPRequest = ""
+	task.HTTPResponseCode = nil
+	task.ID = ""
+	task.Summary = nil
+}
+
+func (o *CruiseControlOperation) GetCurrentTask() *CruiseControlTask {
```

Comment:
Thank you! Good to know. In kubernetes GetName() ... are used. Im not sure which would be the best. Follow the go pattern or follow the kubernetes :)

Reply:
TBH I'm heavily in favor of the Go by default.

Isn't K8s follow the Get... pattern where it does resource CRUD operations due to the different action verbs you can do from RBAC standpoint on resources? In that case Get makes sense, because it signals an HTTP GET which is fine to use Get prefix even in Go, it is just regular field accessors are ugly in Go with Get prefix and are considered unidiomatic.